media/server/gstplayer/source/GstWebAudioPlayer.cpp

```diff
@@ -114,42 +133,247 @@ GstWebAudioPlayer::~GstWebAudioPlayer()
 
     m_gstDispatcherThread.reset();
 
-    // TODO(RIALTO-2): Add shutdown task
-    // Shutdown task thread
-    m_workerThread->join();
-    m_workerThread.reset();
+    resetWorkerThread();
+
+    termWebAudioPipeline();
+}
+
+bool GstWebAudioPlayer::initWebAudioPipeline()
+{
+    m_context.pipeline = m_gstWrapper->gstPipelineNew("webaudiopipeline");
```

Comment:
Minor, but would it be helpful to give each pipeline a unique name perhaps based on 'priority'?

Reply:
Good idea, will do in follow up delivery